### PR TITLE
The Tracers Become Self Aware

### PR DESCRIPTION
**Pull request by ncteisen**

### CODE REVIEW

File: src/core/lib/surface/init.c [hunk 1]
```diff
@@ -120,29 +120,29 @@ void grpc_init(void) {
...
     grpc_mdctx_global_init();
     grpc_channel_init_init();
-    grpc_register_tracer("api", &grpc_api_trace);
-    grpc_register_tracer("channel", &grpc_trace_channel);
-    grpc_register_tracer("connectivity_state", &grpc_connectivity_state_trace);
-    grpc_register_tracer("channel_stack_builder",
+    grpc_register_tracer(&grpc_api_trace);
+    grpc_register_tracer(&grpc_trace_channel);
+    grpc_register_tracer(&grpc_connectivity_state_trace);
+    grpc_register_tracer(
```
Comment (ctiller) [inline src/core/lib/surface/init.c]: I assume clang-format fails here

**Comment by ctiller** [comment]
One last thought: add a tracer "list"... which if enabled prints the list of registered tracers at the end of grpc_init.

Future problem: fail sanity if doc/environment_variables.md is out of sync.

**Comment by ncteisen** [comment]
Added a list_tracer flag, and also threw in a refcount flag that turns on all refcount debugging tracers

**Comment by ncteisen** [comment]
Windows: #11745 
Linux: #10894 
Portability is an infra issue

**Pull request merged**